lib/kafka/connection.rb

```diff
@@ -89,6 +102,12 @@ def request(api_key, request, response_class)
           end
         end
       end
+    rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT => e
+      @logger.error "Connection error: #{e}"
+
+      close
```

Comment:
Woudn't it be good to start new connection setup attempt right away and not to wait for next `request` call?